nx/lib/nx.ex

```diff
@@ -17271,4 +17269,32 @@ defmodule Nx do
     |> log()
     |> add(max)
   end
+
+  @doc """
+  Removes duplicate data from a single dimension tensor.
+
+  ## Examples
+
+      iex> Nx.uniq(Nx.tensor([1,2,3,3,4,5,5,6,7,8]))
+      #Nx.Tensor<
+        s64[8]
+        [1, 2, 3, 4, 5, 6, 7, 8]
+      >
+  """
+  @doc type: :creation
+  def uniq(tensor) do
+    rank_limit = 1
+
+    apply_vectorized(tensor, fn t ->
+      if rank(t) == rank_limit do
+        tensor
+        |> to_list()
+        |> Enum.uniq()
```

Comment:
This is also not a desirable pattern for inclusion in Nx because it can't be used inside defn.

Reply:
Got it. I tried to use MapSets or Common maps, but NX does not support any of this in map/reduce. Which is the recommend way to have "HasMaps" in NX or something I can use to remove duplicate data from the tensor?

Reply:
Not in any meaningful way to what you want, basically because any deduping implies dynamic shapes

Reply:
Got..it maybe a way would be to create the tensor from the beginning with a flag to remove duplicate data... `Nx.tensor(allow_duplicate: false)`it will create a simple vector without duplications.. from that we can use it for anything we want... including one_hot_encoder and label_encoder.

Reply:
How would this option behave with a tensor that has more than 1 axis?

Reply:
Maybe remove duplicate data along all the available axis ?

Reply:
```
[
  [1, 3, 3],
  [3, 1, 2]
]
```

What happens with this tensor in this case?

I don't see a way to define this deduping operation in a general sense without referring to a single axis.

Reply:
And even in this case, this tensor is still problematic if we dedupe along `axis: 1`, because the first row would end up with 2 elements and the second row would end up with 3 elements.

Given this, I believe that this would only apply to rank 1 tensors, in which case a simple Enum.uniq before Nx.tensor would suffice